tests/test_cli.py

```diff
@@ -668,3 +670,71 @@ def mock_operation(*args, **kwargs) -> int:
     
     assert exp_output in captured.out
     assert actual_retval == exp_retval
+
+
+def _good_build(*args, **kwargs):
+    print("LGTM")
+
+def _bad_build(*args, **kwargs):
+    raise Exception
+
+@contextmanager
+def _totally_real_temp_dir(*a, **kw):
+    yield "/path/to/a/real/temp/dir"
+
+@pytest.mark.parametrize(
```

Comment:
I like the compactness of the test, but it took me a couple of minutes to figure out to determine the validity of the test because of the monkeypatches. Please expand out the docstring for this test a bit.